src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -4862,6 +4863,93 @@ bool QgsArrayForeachExpressionFunction::prepare( const QgsExpressionNodeFunction
   return true;
 }
 
+QgsArrayFilterExpressionFunction::QgsArrayFilterExpressionFunction()
+  : QgsExpressionFunction( QStringLiteral( "array_filter" ), 2, QCoreApplication::tr( "Arrays" ) )
+{
+
+}
+
+bool QgsArrayFilterExpressionFunction::isStatic( const QgsExpressionNodeFunction *node, QgsExpression *parent, const QgsExpressionContext *context ) const
+{
+  bool isStatic = false;
+
+  QgsExpressionNode::NodeList *args = node->args();
+
+  if ( args->count() < 2 )
+    return false;
+
+  if ( args->at( 0 )->isStatic( parent, context ) && args->at( 1 )->isStatic( parent, context ) )
+  {
+    isStatic = true;
+  }
+  return isStatic;
+}
+
+QVariant QgsArrayFilterExpressionFunction::run( QgsExpressionNode::NodeList *args, const QgsExpressionContext *context, QgsExpression *parent, const QgsExpressionNodeFunction *node )
+{
+  Q_UNUSED( node )
+  QVariantList result;
+
+  if ( args->count() < 2 )
+    // error
+    return result;
+
+  const QVariantList array = args->at( 0 )->eval( parent, context ).toList();
+
+  QgsExpressionContext *subContext = const_cast<QgsExpressionContext *>( context );
+  if ( !context )
+    subContext = new QgsExpressionContext();
```

Comment:
Could you do https://github.com/qgis/QGIS/pull/7685/commits/416dbf654734f450cc55ff6f3f50033f11b4dd5a approach to avoid the raw pointers?